src/cargo/ops/registry.rs

```diff
@@ -470,31 +484,30 @@ fn wait_for_publish(
                 }
             };
             if !summaries.is_empty() {
-                break;
+                break true;
             }
         }
 
-        if timeout < now.elapsed() {
+        let elapsed = now.elapsed();
+        if timeout < elapsed {
             config.shell().warn(format!(
-                "timed out waiting for `{}` to be in {}",
-                pkg.name(),
-                source_description
+                "timed out waiting for `{short_pkg_description}` to be available in {source_description}",
             ))?;
-            break;
-        }
-
-        if !logged {
-            config.shell().status(
-                "Waiting",
-                format!(
-                    "on `{}` to propagate to {} (ctrl-c to wait asynchronously)",
-                    pkg.name(),
-                    source_description
-                ),
+            config.shell().note(
+                "The registry may have a backlog that is delaying making the \
+                crate available. The crate should be available soon.",
             )?;
-            logged = true;
+            break false;
         }
+
+        progress.tick_now(elapsed.as_secs() as usize, max, "")?;
         std::thread::sleep(sleep_time);
+    };
+    if is_available {
+        config.shell().status(
+            "Completed",
+            format!("{pkg} has been successfully published to {source_description}"),
```

Comment:
This reads funny considering we already said it was published

Reply:
Yea, I'm happy to try to come up with better wording. I agree this didn't hit a sweet spot of being short and clear. Do you have any suggestions? I'd like to have some kind of final statement that indicates everything is complete.

Reply:
The given motivation for this is:

> Provides more of a description of why Cargo is waiting to try to make it clearer what is happening.

Before we had this part of the "Waiting" comment and it said `(ctrl-c to wait asynchronously)`.  Besides trying to expand, I'm assuming a part of this is we can't append this to the "Waiting" or put it after because of the progress bars.


The full text is
```
note: Waiting up to 60 seconds for `foo v0.0.1` to be available at registry `crates-io`.
You may press ctrl-c to skip waiting; the crate should be available shortly.
```

I feel like this includes details that aren't important enough to clog up the screen by default (like the timeout time) and a lot of this just repeats what we are saying in other messages.  If we could just mention the ctrl-c as a note *after* the Waiting, that would be a good way to get this align but I suspect the progress bar doesn't like that?

Something like
```
Uploaded foo v0.0.1 ([..]) to registry `crates-io`
     Waiting [===>                       ] 11/60
note: You may press ctrl-c to skip waiting; the crate should be available shortly.
Published foo v0.0.1 ([..]) at registry `crates-io`
```